docs/scripts/util/crawlAllLinksFromAllPages.ts

```diff
@@ -0,0 +1,41 @@
+import puppeteer from 'puppeteer';
+import { runArrayPromiseInOrder } from '.';
+
+export async function crawlAllLinksFromAllPages(allPages: string[]) {
+  type AllPagesPaths = Map<
+    string,
+    {
+      pageUrl: string;
+      links: { href: string; tagName: string; tagText: string }[];
+    }
+  >;
+  const allPagesPaths: AllPagesPaths = new Map();
+  await runArrayPromiseInOrder(allPages, checkSitemapPath);
+
+  return allPagesPaths;
+
+  async function checkSitemapPath(pageUrl: string, pageIdx: string) {
+    let browser = await puppeteer.launch();
+
+    const page = await browser.newPage();
+
+    await page.goto(pageUrl, { waitUntil: 'networkidle2' });
+    await page.waitForNetworkIdle({ idleTime: 1000 });
+    console.log(`🐝 [VISITING...] page #${pageIdx} ${pageUrl}`);
+    await page.setViewport({ width: 1080, height: 1024 });
+    const links = await page.evaluate(() => {
+      return Array.from(document.querySelectorAll('a')).map(
+        ({ href, tagName, text }) => ({
+          href,
+          tagName,
+          tagText: text,
+        })
+      );
+    });
```

Comment:
what is the advantage of using `page.evaluate` and `document.querySelectorAll` over:
```
const links = await page.$$('a');
...
... await link.evaluate(a => a.href, ...
```
ref: `https://pptr.dev/api/puppeteer.page.__`

Reply:
Good call. I didn't compare them and just started with `page.evaluate`....is there any advantage of using `page.$$`?

Reply:
It would be like this

```ts
    const linkElements = await page.$$('a');

    const links = [];
    await runArrayPromiseInOrder(linkElements, async (link) => {
      const linkVal = await link.evaluate(({ href, tagName, text }) => ({
        href,
        tagName,
        tagText: text,
      }));
      links.push(linkVal);
    });

    allPagesPaths.set(pageIdx, { pageUrl, links });
```

so I think `page.evaluate` looks a lot cleaner